### PR TITLE
Allow sort by on all fields in MappedInstances.tsx

### PR DESCRIPTION
**Pull request by RNHTTR**
It's difficult to find problematic TIs when attempting to debug tasks that have dozens or hundreds of mapped TIs. For example, imagine 99 mapped instances completed in 10 seconds but 1 mapped instance took 10 minutes. It's like finding a needle in a haystack to find that one. 

This will allow users to sort by each field in the grid view.

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by bbovenzi** [comment]
The sort for those fields is disabled because the REST API only supports map index and state right now. We would need to update it there too. See [here](https://github.com/apache/airflow/blob/main/airflow/api_connexion/endpoints/task_instance_endpoint.py#L242) for API code

File: airflow/api_connexion/endpoints/task_instance_endpoint.py [hunk 1]
```diff
@@ -239,16 +239,32 @@ def get_mapped_task_instances(
...
+        elif order_by == "-end_date":
+            entry_query = entry_query.order_by(TI.end_date.desc(), TI.map_index.asc())
         elif order_by == "-map_index":
+            print("map index!!")
             entry_query = entry_query.order_by(TI.map_index.desc())
         else:
             raise BadRequest(detail=f"Ordering with '{order_by}' is not supported")
     else:
+        print("map inde ascx!!")
         entry_query = entry_query.order_by(TI.map_index.asc())
```
Comment (uranusjr) [inline airflow/api_connexion/endpoints/task_instance_endpoint.py]: We can flatten this to

```python
if order is None:
    entry_query = entry_query.order_by(TI.map_index.asc())
elif order_by == "state":
    entry_query = entry_query.order_by(TI.state.asc(), TI.map_index.asc())
... # More...
```

**Review by uranusjr — APPROVED** [review]

File: airflow/www/static/js/dag/details/taskInstance/MappedInstances.tsx [hunk 2]
```diff
@@ -47,7 +47,12 @@ const MappedInstances = ({ dagId, runId, taskId, onRowClicked }: Props) => {
   const sort = sortBy[0];
 
   const orderBy =
-    sort && (sort.id === "state" || sort.id === "mapIndex")
+    sort &&
+    (sort.id === "state" ||
+      sort.id === "mapIndex" ||
+      sort.id === "duration" ||
+      sort.id === "startDate" ||
+      sort.id === "endDate")
```
Comment (bbovenzi) [inline airflow/www/static/js/dag/details/taskInstance/MappedInstances.tsx]:
```suggestion
    sort && sort.id
```
We can simplify this check now.

**Review by bbovenzi — APPROVED** [review]
lgtm. just one nitpick

**Pull request merged**